draft-ietf-tls-tls13.md

```diff
@@ -2376,7 +2376,15 @@ RSASSA-PSS algorithms
   digest used in the mask generation function and the digest being signed are
   both the corresponding hash algorithm as defined in {{!SHS}}. When used in
   signed TLS handshake messages, the length of the salt MUST be equal to the
-  length of the digest output.  This codepoint is new in this document and is also
+  length of the digest output. When used in certificate signatures, the
+  length of the salt MUST be at least as long as the length of the
+  digest output, and the algorithm parameters MUST be DER encoded.
```

Comment:
I'm not sure that this is the appropriate document to assert such a MUST. What can a TLS implementation do if a CA document allows including BER parameters?

Reply:
I don't think BER is allowed anywhere in X.509

Reply:
It is.  Look for the word BER in the PKIX RFC 5280.

Reply:
@richsalz the only reference to BER I see in 5280 is:

```
   Where the information is available via HTTP or FTP, accessLocation
   MUST be a uniformResourceIdentifier and the URI MUST point to either
   a single DER encoded certificate as specified in [RFC2585] or a
   collection of certificates in a BER or DER encoded "certs-only" CMS
   message as specified in [RFC2797].
```

so that reiterates it - if you talk about single certificate, it MUST be DER encoded...

Reply:
I don't think that they are allowed, it was that just some CAs were creating malformed (BER encoded) certificates and so support for it was added/ossified.

I have a vague recollection of seeing somewhere among the PKCS#12 related RFCs that any structure that is signed must be DER encoded...